packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts

```diff
@@ -333,8 +333,14 @@ export class AnnotationsStateMachine extends StateMachine<States, AnnotationType
 
                 dblclick: {
                     guard: guardActiveHasLineText,
-                    action: () => {
-                        ctx.showAnnotationSettings(this.active!, undefined);
+                    action: ({ offset }) => {
+                        const nodeAtCoords = ctx.getNodeAtCoords(offset, this.active!);
+
+                        if (!(nodeAtCoords === 'line' || nodeAtCoords === 'text')) {
```

Comment:
```suggestion
                        if (nodeAtCoords !== 'line' && nodeAtCoords !== 'text') {
```

Generally, it's a bit more readable like this, as the `(!(` can be missed when scanning the code.

Reply:
Also, I'd guess we will want to open on the "line" tab when the user double clicks a handle? So could do something like `const node = nodeAtCoords === 'text' ? 'text' : 'line';`